text/todo-generic-payload.md

```diff
@@ -0,0 +1,368 @@
+- `DISCRETE`: Digital controls with enumerated values (modes, frequencies)
+
+### Message ID Allocation
+
+Proposed message IDs in the 60000-60004 range (vendor-specific range) to avoid conflicts during development. Final standardization would require official MAVLink message ID allocation.
+
+### Extension Fields
+
+All messages include extension fields for future enhancements:
+
+- Power consumption monitoring
+- Temperature sensing
+- Mass/inertia properties
+- 64-bit value support
+
+## Alternatives
+
+### Alternative 1: Extend Existing Payload Commands
+
+Build upon the current [MAVLink payload services](https://mavlink.io/en/services/payload.html) by adding more generic commands and standardizing implementation.
+
+**Analysis**: The existing `MAV_CMD_DO_SET_ACTUATOR` and `MAV_CMD_DO_SET_SERVO` commands could theoretically be extended with additional parameters for type information and metadata.
+
+**Key Limitations**:
+
+- Commands are stateless and don't provide discovery mechanisms
+- Inconsistent implementation across flight stacks ("only implemented on PX4")
+- Limited parameter space in command messages prevents rich metadata
+- No built-in telemetry streaming capabilities
+- Would require significant breaking changes to existing command semantics
+
+### Alternative 2: Parameter-Only Approach
+
+Use MAVLink's parameter system (`PARAM_*` messages) to expose payload functions as parameters.
+
+**Analysis**: Parameters provide get/set functionality and some metadata (min/max, type). Could potentially expose payload functions as settable parameters.
+
+**Key Limitations**:
+
+- Parameters are designed for configuration, not real-time control
+- No standardized discovery of which parameters belong to payloads
+- Limited data types (mostly float32)
+- No support for momentary operations or complex control modes
+- Parameter namespace pollution with payload-specific entries
+- No built-in telemetry streaming for sensor data
+
+### Alternative 3: Extend Component Information Service
+
+Build upon the [MAVLink Component Information Service](https://mavlink.io/en/services/component_information.html) to include payload metadata and add control messages.
+
+**Analysis**: The component information service provides sophisticated metadata discovery through JSON files and supports actuator configuration. This could potentially be extended to describe payload capabilities.
+
+**Key Limitations**:
+
+- Currently "work in progress" with limited adoption
+- Designed for static configuration metadata, not dynamic control
+- Requires external file hosting or on-device storage for metadata
+- JSON parsing overhead for simple operations
+- No real-time control or telemetry streaming mechanisms
+- Metadata assumed invariant after boot, limiting dynamic payloads
+
+## Unresolved Questions
+
+1. **Message ID Range**: Should we request official MAVLink message IDs or remain in vendor-specific range?
```

Comment:
This would go into development.xml with official message ids in the common range IFF RFC is broadly accepted. If not accepted you could still implement this in your own dialect.